riscv/insns/fadd_d.h

```diff
@@ -1,5 +1,6 @@
 require_either_extension('D', EXT_ZDINX);
 require_fp;
+require_stateen_fcsr;
```

Comment:
Can this check be part of `require_fp`? The Smstateen spec says:

> For convenience, when the stateen CSRs are implemented and misa.F = 0, then if bit 1 of a controlling stateen0 CSR is zero, all floating-point instructions cause an illegal instruction trap (or virtual instruction trap, if relevant), as though they all access fcsr, regardless of whether they really do.

Reply:
It's OK to take it into require_fp from the perspective of function realization. However, I'm not sure whether the logic of stateen check which is to control the access of FCSR is suitable for the original logic of require_fp.
.